### ISSUE TITLE
InvalidCastException on TermsAggregate deserialization

### ISSUE THREAD

**Issue by Chakrygin** [elastic/elasticsearch-net#2531]
**NEST/Elasticsearch.Net version**: 5.0.0

**Elasticsearch version**: 5.1.1

**Steps to reproduce**:
 1. Test data:

```
POST http://localhost:9200/_bulk

{"index":{"_index":"test","_type":"test","_id":"1"}}
{"id":1,"gid":4,"availability":1}
{"index":{"_index":"test","_type":"test","_id":"2"}}
{"id":2,"gid":4,"availability":2}
{"index":{"_index":"test","_type":"test","_id":"3"}}
{"id":3,"gid":5,"availability":1}
```

 2. C# code:

```
var elastic = new ElasticClient(
	new Uri("http://ipv4.fiddler:9200/"));

var response = elastic.Search<object>(request => request
	.Index("test")
	.Type("test")
	.Aggregations(aggs => aggs
		.Terms("$items", items => items
		    .Field("gid")
		    .Order(new TermsOrder {Key = "availability", Order = SortOrder.Ascending})
		    .Aggregations(innerAggs => innerAggs
		        .Min("availability", agg => agg
		            .Field("availability"))
		        .Max("score", agg => agg
		            .Script("_score"))))));

var result = response.Aggs.Terms<int>("$items");
```

**Describe the feature**:

Unable to cast object of type 'Nest.SignificantTermsBucket' to type 'Nest.KeyedBucket`1[System.Object]'.

**Comment by gmarz** [comment]
The problem stems from the fact that your max aggregation is named `score`, which is also a property of the significant terms aggregation, and thus our parser is getting confused as there's no way currently to distinguish between the two.

While not really a fix, a simple work around would be to name the max aggregation to something other than `score`.

Due to the way we currently parse the aggregation response stream on the fly, I'm not sure of a good way to fix this yet.  We had spiked [this PR](https://github.com/elastic/elasticsearch-net/pull/2081) which used the metadata field to inject the aggregation type which made de-serialization deterministic, but decided to not pull it in as we felt it was too invasive/surprising to the user.  However, it's purpose was to prevent problems exactly like this.

**Comment by Chakrygin** [comment]
Thank you. Renaming solves this problem.

**Comment by gmarz** [comment]
Closed via #2551
